src/MatrixParallel/MatrixParallel/Matrix.cs

```diff
@@ -0,0 +1,209 @@
+using System;
+using System.IO;
+using System.Threading;
+ 
+namespace MatrixParallel
+{
+    /// <summary>
+    /// Class for matrix object and operations on it
+    /// </summary>
+    public class Matrix
+    {
+        private int Rows { get; set; }
+        private int Cols { get; set; }
+        private long[,] Matr { get; set; }
+ 
+        /// <summary>
+        /// Constructor for Matrix by given parameters
+        /// </summary>
+        /// <param name="rows"></param>
+        /// <param name="cols"></param>
+        /// <exception cref="ArgumentOutOfRangeException"></exception>
+        public Matrix(int rows, int cols)
+        {
+            if (rows <= 0)
+            {
+                throw new ArgumentOutOfRangeException(nameof(rows), "Number of rows should be positive");
+            }
+            if (cols <= 0)
+            {
+                throw new ArgumentOutOfRangeException(nameof(cols), "Number of columns should be positive");
+            }
+            
+            var matr = new long [rows, cols];
+            Rows = rows;
+            Cols = cols;
+            Matr = matr;
+        }
+
+        /// <summary>
+        /// 2dArray to an object "Matrix"
+        /// </summary>
+        /// <param name="matrix"></param>
+        public Matrix(long[,] matrix)
+        {
+            Rows = matrix.GetLength(0);
+            Cols = matrix.GetLength(1);
+            Matr = matrix;
+        }
+ 
+        /// <summary>
+        /// Generates a Matrix with random elements by given parameters 
+        /// </summary>
+        /// <param name="rows"></param>
+        /// <param name="columns"></param>
+        /// <returns></returns>
+        public static Matrix Generate (int rows, int columns)
+        {
+            var random = new Random();
+            var resMatrix = new long [rows, columns];
+            
+            for (int i = 0; i < rows; i++)
+            {
+                for (int j = 0; j < columns; j++)
+                {
+                    resMatrix[i, j] = random.Next(50);
+                }
+            }
+            var result = new Matrix(resMatrix);
+            return result;
+        }
+ 
+        /// <summary>
+        /// Constructor that gets a matrix from given file
+        /// </summary>
+        /// <param name="filePath"></param>
+        public Matrix (string filePath)
+        {
+            string[] lines = File.ReadAllLines(filePath);
+            int standLength = lines[0].TrimEnd(' ').Split(' ').Length;
+
+            long[,] matrix = new long [lines.Length, standLength];
```

Comment:
```suggestion
            var matrix = new long [lines.Length, standLength];
```
В современном C# вообще не надо писать тип дважды в одной строке